src/DependencyInjection/Compiler/ModelManagerCompilerPass.php

```diff
@@ -26,22 +27,28 @@
  */
 final class ModelManagerCompilerPass implements CompilerPassInterface
 {
+    public const MANAGER_TAG = 'sonata.admin.manager';
+
     public function process(ContainerBuilder $container): void
     {
         $availableManagers = [];
 
-        foreach ($container->getServiceIds() as $id) {
-            if (0 !== strpos($id, 'sonata.admin.manager.') || !is_subclass_of($container->getDefinition($id)->getClass(), ModelManagerInterface::class)) {
-                continue;
+        foreach ($container->findTaggedServiceIds(self::MANAGER_TAG) as $id => $tags) {
+            $definition = $container->getDefinition($id);
+
+            if (!is_subclass_of($definition->getClass(), ModelManagerInterface::class)) {
+                throw new LogicException(sprintf('Service "%s" tagged with "sonata.admin.manager" must implement `%s`.', $id, ModelManagerInterface::class));
             }
 
-            $availableManagers[$id] = $container->getDefinition($id);
+            $availableManagers[$id] = $definition;
         }
 
-        $bundles = $container->getParameter('kernel.bundles');
-        if (isset($bundles['MakerBundle'])) {
-            $adminMakerDefinition = $container->getDefinition('sonata.admin.maker');
-            $adminMakerDefinition->replaceArgument(1, $availableManagers);
+        if ($availableManagers) {
```

Comment:
```suggestion
        if (count($availableManagers) > 0) {
```

Reply:
Could I use `!empty($availableManagers)` instead?

Reply:
Should be okay. 

I don't like `empty` at all, because it will check for null, empty string or empty array.

Reply:
I agree, but `count()` doesn't work with iterators neither.